### ISSUE TITLE
mmkdir test fails when MANTA_TEST_ROLE is unset

### ISSUE THREAD

**Issue by bahamas10** [joyent/node-manta#347]
I ran the tests without `MANTA_TEST_ROLE` defined and the `mmkdir` test failed as it tried to delete a directory that was never created.

Before

```
$ ./node_modules/.bin/nodeunit test/mmkdir.test.js 

mmkdir.test.js
✖ cleanup: rm test directory /dave/stor/node-manta-test-mput-513714de

VError: exec "/home/dave/joyent/node-manta/bin/mrm -r /dave/stor/node-manta-test-mput-513714de": exited with status 1
    at interpretChildProcessResult (/home/dave/joyent/node-manta/node_modules/forkexec/lib/forkexec.js:163:15)
    at onChildExited (/home/dave/joyent/node-manta/node_modules/forkexec/lib/forkexec.js:75:9)
    at /home/dave/joyent/node-manta/node_modules/forkexec/lib/forkexec.js:53:4
    at ChildProcess.exithandler (child_process.js:282:5)
    at emitTwo (events.js:126:13)
    at ChildProcess.emit (events.js:214:7)
    at maybeClose (internal/child_process.js:925:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)


FAILURES: 1/1 assertions failed (1435ms)
```

After

```
$ ./node_modules/.bin/nodeunit test/mmkdir.test.js 

OK: 0 assertions (108ms)
```

**Comment by bahamas10** [comment]
https://cr.joyent.us/#/c/4706/

**Comment by kellymclaughlin** [comment]
Nice catch. I fixed this as part of the CR for #346 as well, but might as well have it as a separate commit.
